src/Analyzers/CSharp/CodeFixes/UseInterpolatedString/CSharpUseInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.UseInterpolatedString;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseInterpolatedString;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseInterpolatedString), Shared]
+[method: ImportingConstructor]
+[method: SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+internal partial class CSharpUseInterpolatedStringCodeFixProvider() : AbstractUseInterpolatedStringCodeFixProvider
+{
+    protected override async Task FixOneAsync(Document document, Diagnostic diagnostic, SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        var span = diagnostic.Location.SourceSpan;
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var node = root.FindNode(span, getInnermostNodeForTie: true) as LiteralExpressionSyntax;
+        if (node?.Span != span || !node.IsKind(SyntaxKind.StringLiteralExpression))
+            return;
+
+        var newNode = CreateInterpolatedString(node);
+        editor.ReplaceNode(node, newNode);
+    }
+
+    private static InterpolatedStringExpressionSyntax CreateInterpolatedString(SyntaxNode stringLiteralExpressionNode)
+    {
+        var token = stringLiteralExpressionNode.GetFirstToken();
+        var originalTokenKind = token.Kind();
+        var isVerbatimString = token.IsVerbatimStringLiteral();
+        var newTokenKind = GetKind(originalTokenKind);
+        var startTokenKind = GetStartToken(originalTokenKind, isVerbatimString);
+        var text = GetTextWithoutQuotes(token, isVerbatimString);
+        var endTokenKind = GetEndToken(originalTokenKind);
+
+        var newNode = SyntaxFactory.InterpolatedStringText(
+            SyntaxFactory.Token(
+                leading: SyntaxTriviaList.Empty,
+                kind: newTokenKind,
+                text: text,
+                valueText: text,
+                trailing: SyntaxTriviaList.Empty))
+            .WithTriviaFrom(stringLiteralExpressionNode);
```

Comment:
Isn't there an overload of `InterpolatedStringText`, that accepts leading and trailing trivia, so we can avoid unnecessary intermidiate node allocation?

Reply:
Nope!
![image](https://github.com/dotnet/roslyn/assets/7676354/f1ba77a1-3dcd-4c78-8847-6fa3f810ff19)